### DISCUSSION TITLE
How do you add plugins to chart widget?

### DISCUSSION THREAD

**Discussion by taqillion** [Laravel-Backpack/community-forum#689] [Q&A (Help)] [answered]
I am using the Backpack Chart Widget which uses the chartjs by default.
Now chartjs has some plugins that I would like to utilize like [data labels](https://chartjs-plugin-datalabels.netlify.app/) and [zoom](https://www.chartjs.org/chartjs-plugin-zoom/latest/)

How do I configure my Chart Widget to use those plugins?

**Comment by tabacitu** [comment, marked as answer]
Hi @A-Taqi ,

There is no built-in way for you to use those plugins. But you can create a custom Chart widget, based on ours. Here are some quick steps and pseudocode from the back of my head:

---
 
**Solution A.** Create a new blade file (eg. `resources/views/vendor/backpack/ui/widgets/custom-chart.blade.php`). Inside it, include our blade file, then at the end add a section to push any scripts or changes you want:
```php
@include('backpack.pro::widgets.chart')

@push('after_scripts')
<script>
    // TODO: your custom code
</script>
@endpush
```
Then you can use your custom chart widget, instead of the one provided by PRO:
```php
    Widget::add([
        'type'    => 'custom-chart',
        'wrapper' => ['class' => 'col-sm-12'],
        // ...
    ]);
```

---

**Solution B.** Just copy the blade file from the repo, to your `resources` directory, and edit it in any way you want. Make any changes you need to that widget, including adding those plugins. 

---

I recommend solution A, because you'll also be getting any updates we push to that blade file. But if that's not possible, option B should be good enough too.

Hope it helps!
